src/v/cloud_storage/topic_path_utils.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "base/seastarx.h"
+#include "cloud_storage/remote_label.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+
+#include <seastar/core/sstring.hh>
+
+namespace cloud_storage {
```

Comment:
Is it worth leaving a comment explaining/clarifying the distinction between the "labeled" (new) vs "prefixed" (old) style in the code here and elsewhere for `topic`/`partition`/`segment` path utils?

I can imagine that most people won't know what is legacy and what is new from these function names.

Reply:
This is a great point. Will do this in a follow-up PR.

Reply:
Yeh, I'd be into this as well. I was just kinda guessing about the distinction.